modules/precisoBidAdapter.js

```diff
@@ -0,0 +1,513 @@
+import {getOrigin} from '../libraries/getOrigin/index.js';
+import {BANNER, NATIVE, VIDEO} from '../src/mediaTypes.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {includes} from '../src/polyfill.js';
+import {convertOrtbRequestToProprietaryNative} from '../src/native.js';
+import {config} from '../src/config.js';
+
+const BIDDER_CODE = 'preciso';
+const REGIONS = ['prebid-eu', 'prebid-us', 'prebid-asia'];
+const ENDPOINT_URL = 'mndtrk.com/bidder/prebid/bids';
+const FLEDGE_ENDPOINT_URL = 'mndtrk.com/bidder/prebidfledge/bids';
+const DEFAULT_CURRENCY_ARR = ['USD']; // NOTE - USD is the only supported currency right now; Hardcoded for bids
+const SUPPORTED_MEDIA_TYPES = [BANNER, NATIVE, VIDEO];
+const TTL = 100;
+const GVLID = 874;
+
+// Codes defined by OpenRTB Native Ads 1.1 specification
+export const OPENRTB = {
+  NATIVE: {
+    IMAGE_TYPE: {
+      ICON: 1,
+      MAIN: 3,
+    },
+    ASSET_ID: {
+      TITLE: 1,
+      IMAGE: 2,
+      ICON: 3,
+      BODY: 4,
+      SPONSORED: 5,
+      CTA: 6
+    },
+    DATA_ASSET_TYPE: {
+      SPONSORED: 1,
+      DESC: 2,
+      CTA_TEXT: 12,
+    },
+  }
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: SUPPORTED_MEDIA_TYPES,
+  gvlid: GVLID,
+
+  isBidRequestValid: function (bid) {
+    return !!(includes(REGIONS, bid.params.region) && bid.params.publisherId);
+  },
+  buildRequests: function (validBidRequests, bidderRequest) {
+    // convert Native ORTB definition to old-style prebid native definition
+    validBidRequests = convertOrtbRequestToProprietaryNative(validBidRequests);
+
+    const request = {
+      id: bidderRequest.bidderRequestId,
+      imp: validBidRequests.map(slot => mapImpression(slot, bidderRequest)),
+      site: mapSite(validBidRequests, bidderRequest),
+      cur: DEFAULT_CURRENCY_ARR,
+      test: validBidRequests[0].params.test || 0,
+      source: mapSource(validBidRequests[0], bidderRequest),
+    };
+
+    if (bidderRequest && bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies) {
+      const consentStr = (bidderRequest.gdprConsent.consentString)
+        ? bidderRequest.gdprConsent.consentString.replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') : '';
+      const gdpr = bidderRequest.gdprConsent.gdprApplies ? 1 : 0;
+      request.regs = {ext: {gdpr: gdpr}};
+      request.user = {ext: {consent: consentStr}};
+    }
+    if (validBidRequests[0].schain) {
+      const schain = mapSchain(validBidRequests[0].schain);
+      if (schain) {
+        request.ext = {
+          schain: schain,
+        };
+      }
+    }
+
+    if (validBidRequests[0].userIdAsEids) {
+      const eids = { eids: validBidRequests[0].userIdAsEids };
+      if (request.user && request.user.ext) {
+        request.user.ext = { ...request.user.ext, ...eids };
+      } else {
+        request.user = {ext: eids};
+      }
+    }
+
+    const ortb2Params = bidderRequest?.ortb2 || {};
```

Comment:
Could you update to use `getConfig('ortb2')` to access this data?

Reply:
Taking the rtbhouseBidAdapter.js as a reference, the precisoBidAdapter.js was created. We noticed that getConfig('ortb2') is not used in the rtbhouseBidAdapter.js file. Could you please explain how to use getConf('ortb2)?

Reply:
More information can be found at the documentation, it retrieves ortb2 data from the auction: https://docs.prebid.org/dev-docs/publisher-api-reference/setConfig.html#setConfig-fpd

Reply:
@PrecisoSRL We generally ask new adapters to use the getConfig method for pulling ortb2 data, not to directly access it from the bidrequest. Can you update, this should be no downstream change to the data you receive.